Sources/KlaviyoSwift/Klaviyo.swift

```diff
@@ -464,288 +336,28 @@ public class Klaviyo: NSObject  {
-            let characterSet = NSMutableCharacterSet.urlQueryAllowed
-            b64String = data!.base64EncodedString()
-            b64String = b64String.addingPercentEncoding(withAllowedCharacters: characterSet)!
-        }
-        
-        return b64String
-    }
-    
-    /**
-     JSONSerializeObject: serializes an AnyObject into an NSData Object.
-     
-     - Parameter obj: the object to be serialized
-     - Returns: NSData representation of the object
-     */
-    private func JSONSerializeObject(obj : AnyObject)-> NSData? {
-        
-        let coercedobj = JSONSerializableObjectForObject(obj: obj)
-        var error : NSError? = nil
-        var data : NSData? = nil
-        
-        if JSONSerialization.isValidJSONObject(obj) {
-            do {
-                data = try JSONSerialization.data(withJSONObject: coercedobj, options: .prettyPrinted) as NSData
-            }
-            catch let errors as NSError{
-                data = nil
-                error = errors
-                print("exception encoding the api data: \(errors)")
-            }
-            catch {
-                print("unknown error")
-            }
-            if error == nil { return data }
-            print("Error parsing the json: \(String(describing: error))")
-        }
-        
-        return data
-    }
-    
-    /**
-     JSONSerializableObjectForObject: Function checks & converts data into approved data types for JSON
-     
-     :param: obj type AnyObject
-     :returns: an AnyObject that has been verified and cleaned
-     */
-    private func JSONSerializableObjectForObject(obj: AnyObject)-> AnyObject {
-        
-        if JSONSerialization.isValidJSONObject(obj) {
-            return obj
-        }
-        
-        // Apple Documentation: "All objects need to be instances of NSString, NSNumber, NSArray, NSDictionary, or NSNull"
-        if (obj is NSString || obj is NSNumber || obj is NSNull) {
-            return obj
-        }
-        
-        // recurse through collections and serialize each object
-        if (obj is NSArray) {
-            let a = NSMutableArray()
-            let objects = obj as! NSArray
-            for item in objects {
-                a.add(_: JSONSerializableObjectForObject(obj: item as AnyObject))
-            }
-            return a as NSArray
-        }
-        
-        if (obj is NSDictionary) {
-            let dict = NSMutableDictionary()
-            let objects = obj as! NSDictionary
-            
-            for key in objects.keyEnumerator() {
-                guard let stringKey = key as? String else {
-                    print("warning: property keys should be strings. got: \(String(describing: key))")
-                    continue
-                }
-                let v : AnyObject = JSONSerializableObjectForObject(obj: objects[stringKey]! as AnyObject)
-                dict[stringKey] = v
+    private func dispatchOnMainThread(action: KlaviyoAction) {
```

Comment:
Sorry if I missed an explanation somewhere, but just curious what led to needing to specify main thread for some actions?

Reply:
The store implementation I'm using requires main thread. Eventually I think we can move it to our own thread but the intention is that none of the non-side effect is very heavy weight so this should be ok.

Reply:
This is just a protection against someone calling our sdk from another thread.